pkg/ingester/ingester.go

```diff
@@ -916,6 +938,9 @@ func (i *Ingester) pushSamplesToAppender(userID string, timeseries []mimirpb.Pre
 			stats.sampleOutOfBoundsCount += len(ts.Samples)
 
 			updateFirstPartial(func() error {
+				if ephemeral {
+					return newEphemeralIngestErrSampleTimestampTooOld(model.Time(ts.Samples[0].TimestampMs), ts.Labels)
```

Comment:
[nit] Have you considered passing `ephemeral bool` to `newIngestErrSampleTimestampTooOld()`? I think it could simplify a bit the code.

Reply:
> [nit] Have you considered passing `ephemeral bool` to `newIngestErrSampleTimestampTooOld()`? I think it could simplify a bit the code.

I did exactly that in my first version of the code (not just for `newIngestErrSampleTimestampTooOld` but other `newIngestErr...` methods too). I decided against that approach -- `if ephemeral` was still there, just at different place. And adding new parameter to existing functions broke them at many places (mostly in tests). Adding new functions instead allowed to to keep existing code unchanged, and only add new test cases where appropriate.

Reply:
Ok. I'm good to keep it as is then :)